### ISSUE TITLE
Support other class-based test engines in junitPlatform

### ISSUE THREAD

**Issue by oehme** [gradle/gradle#4252]
Building on #4250, support other class-based engines as well. If they already work, add some test coverage to prove it.

**Comment by raniejade** [comment]
Just FYI, I wouldn't consider Spek class-based. Once https://github.com/Kotlin/KEEP/issues/75 lands, the wrapping class may go away in favor of script files.

**Comment by oehme** [comment]
Got it, in that case #4773 would be the appropriate issue.

**Comment by marcphilipp** [comment]
I've just merged a change yesterday to support parallel test execution which completely overhauled the test execution listener used by Gradle to hook into JUnit Platform events. That should make it much easier to address additional issues.

I think #4912 can be addressed next, and then #4438. So, these things are on our radar and should be fixed in Gradle core but are not the top priority at the moment. Other issues, such as #4773, that would require substantial changes to the whole testing infrastructure could benefit from custom short-term solutions. For example, it's quite easy to [create a custom task that uses the JUnit Platform `ConsoleLauncher`](https://github.com/junit-team/junit5/blob/7f04a28b136e8545206b66bf4b200271805beced/documentation/documentation.gradle.kts#L22-L35). The downside is the missing HTML reporting.

**Comment by naveenchandra91** [comment]
Any update on this issue..please let us know what needs to be done to support this?

**Comment by marcphilipp** [comment]
@naveenchandra91 Class-based engines should generally work. Which one are you struggling with?

**Comment by jjohannes** [comment]
Related issues:
- https://github.com/gradle/gradle/issues/4438
- https://github.com/gradle/gradle/issues/4912

**Comment by ljacomet** [comment]
Sorry for the late reply

Given the number of changes done to JUnit 5 integration, considering this obsolete. If something is still missing, open a new issue against a recent Gradle version with a specific description of the problem and a reproducer.
